doc/ensemble.rst

```diff
@@ -0,0 +1,739 @@
+following arguments are offered:
+
+-  `name`  (str) : key to store function under
+-  `function` (str, optional) : Name of the TorchScript function
+-  `device`  (str, optional) : device for script execution, defaults to “CPU”
+-  `devices_per_node` (int) : assign the number of CPU's or GPU's to use on the node
+
+Example: Loading an TensorFlow Function to the Ensemble
+-------------------------------------------------------
+In this example, we demonstrate how to instruct SmartSim to load
+an TensorFlow function into the database at ensemble runtime. It's
+important to note the function, ``Ensemble.add_function()`` is supported
+for non-colocated deployments and during a colocated deployment, ``Ensemble.add_script()``
+should be used.
+
+**Python Script: Define a TF Function for Ensemble Purposes**
+To load a TF function, define the function within the Python driver script.
+
+.. code-block:: python
+
+    def timestwo(x):
+        return 2*x
+
+**SmartSim Ensemble Integration:**
+
+Assuming an initialized ``Ensemble`` named `smartsim_ensemble`, we specify the
+following parameters to the ``Ensemble.add_function()`` function:
+
+.. code-block:: python
+
+    smartsim_ensemble.add_function(name="example_func", function=timestwo, device="CPU")
+
+In this integration, we provide the following details:
+
+-  `name`: "example_func" - A key to uniquely identify the model within the database.
+-  `function`: timestwo - Name of the TorchScript function defined in the Python driver script.
+-  `device`: "CPU" - Specifying the device for ML model execution.
+
+When the ensemble is started via ``Experiment.start()``, the TF function will be loaded to the
+standard orchestrator that is launched prior to the start of the ensemble.
+
+-------------------
+TorchScript Scripts
+-------------------
+.. _load_script:
+SmartSim supports the execution of TorchScripts
+with each entity belonging to the ensemble. Regardless of whether
+the orchestrator is colocated or not, each script added to the ensemble is
+loaded into the orchestrator prior to the execution of any ensemble member.
+The flexibility of device selection further enhances
+adaptability, offering the choice between "GPU" or "CPU." Users have the flexibility to choose
+between `"GPU"` or `"CPU"` for device selection, and in environments with multiple
+devices, specific device numbers can be specified via `devices_per_node` such as `"GPU:1,"`.
+
+When specifying a TorchScript using ``Ensemble.add_script()``, the
+following arguments are offered:
+
+-  `name`  (str) : key to store script under
+-  `script` (str, optional) : TorchScript code
+-  `script_path` (str, optional) : file path to TorchScript code
+-  `device`  (str, optional) : device for script execution, defaults to “CPU”
+-  `devices_per_node` (int) : assign the number of CPU's or GPU's to use on the node
+
+You might use TorchScript scripts to represent individual models within the ensemble:
```

Comment:
odd wording... TorchScript scripts won't "represent a model" - maybe "You might use TorchScript scripts to _implement_ individual models within the ensemble:"